kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -182,10 +210,12 @@ private Optional<CloseableIterator<FileStatus>> listFromOrNone(
      * Returns the delta files and checkpoint files starting from the given `startVersion`.
      * `versionToLoad` is an optional parameter to set the max bound. It's usually used to load a
      * table snapshot for a specific version.
+     * If no delta files exist below the max bound versionToLoad and at least one delta file exists,
```

Comment:
what does `max bound versionToLoad` mean?

- this (gramatically) makes sense to me: `If no delta files exist below the versionToLoad and at least one delta file exists .... `

why do you need a `delta` file to exist below it? versionToLoad = 11. we have a checkpoint at v10. no delta file at v10. and a delta file at v11?